### PR TITLE
feat(dump): server implementation

### PR DESCRIPTION
**Pull request by melancholiai**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✔                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✖                                                                       |

### CODE REVIEW

**Review by CptSchnitz — CHANGES_REQUESTED** [review]
The server doesn't work if only production dependencies are installed.

**Comment by melancholiai** [comment]
> The server doesn't work if only production dependencies are installed.
fixed in 50e7bf0

File: tests/helpers/index.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  host: 'some_storage_host',
+};
+
+export const DEFAULT_SORT = 'desc';
+
+export const DEFAULT_LIMIT = 10;
+
+export const BOTTOM_FROM = faker.date.past();
+
+export const TOP_TO = faker.date.future(undefined, BOTTOM_FROM);
```
Comment (CptSchnitz) [inline tests/helpers/index.ts:22]: you can run the function without params, as future will always be ahead of the past.

File: tests/helpers/index.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  const { bucket, ...restOfMetadata } = fakeDumpMetadata;
+  const { protocol, host } = mockObjectStorageConfig;
+  return { ...restOfMetadata, url: `${protocol}://${host}/${bucket}/${restOfMetadata.name}` };
+};
+
+export const convertFakesToResponses = (fakeDumpsMetadata: IDumpMetadata[]): DumpMetadataResponse[] => {
+  return fakeDumpsMetadata.map((fake) => convertFakeToResponse(fake));
+};
+
+export const convertToISOTimestamp = (response: DumpMetadataResponse): IntegrationDumpMetadataResponese => {
```
Comment (CptSchnitz) [inline tests/helpers/index.ts:64]: `Response`

File: tests/integration/dumpMetadata/helpers/db.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { container } from 'tsyringe';
+import { Connection, EntityTarget, Repository } from 'typeorm';
+
+import { DumpMetadata } from '../../../../src/dumpMetadata/models/dumpMetadata';
+import { createMultipleFakeDumpsMetadata } from '../../../helpers';
+
+export const getRepositoryFromContainer = <T>(target: EntityTarget<T>): Repository<T> => {
+  const connection = container.resolve(Connection);
+  return connection.getRepository<T>(target);
+};
```
Comment (CptSchnitz) [inline tests/integration/dumpMetadata/helpers/db.ts:10]: this function is only used once, no need to be a separate function.

File: tests/integration/dumpMetadata/helpers/requestSender.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+export function getMockedRepoApp(repo: unknown): Application {
+  container.register('DumpMetadataRepository', { useValue: repo });
+  return getApp();
+}
+
+export async function getDumpsMetadataByFilter(
+  app: Application,
+  filter: DumpMetadataFilter | Record<string, never>,
+  responseType = 'application/json'
```
Comment (CptSchnitz) [inline tests/integration/dumpMetadata/helpers/requestSender.ts:21]: You never this parameter, so its fine to just hardcode it.

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

File: src/common/utils/db.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import { Between, Connection, ConnectionOptions, createConnection, FindOperator } from 'typeorm';
+import { MoreThanOrEqual, LessThanOrEqual } from 'typeorm';
+import { format } from 'date-fns';
+
+import { DumpMetadata } from '../../dumpMetadata/models/dumpMetadata';
+
+export enum DateTypeFormat {
+  DATE = 'yyyy-MM-dd',
+  DATETIME = 'yyyy-MM-dd HH:MM:ss',
+}
```
[the remark this answers is not in the document]
Comment (melancholiai) [reply src/common/utils/db.ts:11]: removed.using postgres for testing also so there is no need for dates formation.

File: src/dumpMetadata/models/dumpMetadata.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+  @Column()
+  public bucket!: string;
+
+  @Index('Timestamp-idx')
+  @Column()
+  public timestamp!: Date;
+
+  @Column('text', { nullable: true })
+  public description!: string;
```
[the remark this answers is not in the document]
Comment (melancholiai) [reply src/dumpMetadata/models/dumpMetadata.ts:27]: done.

File: src/common/utils/db.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+export const moreThanOrEqualTimestamp = (date: Date, type: DateTypeFormat): FindOperator<string> => MoreThanOrEqual(format(date, type));
+
+export const lessThanOrEqualTimestamp = (date: Date, type: DateTypeFormat): FindOperator<string> => LessThanOrEqual(format(date, type));
+
+export const betweenTimestamps = (from: Date, to: Date, type: DateTypeFormat): FindOperator<string> => Between(format(from, type), format(to, type));
```
[the remark this answers is not in the document]
Comment (melancholiai) [reply src/common/utils/db.ts:17]: removed.

File: src/common/utils/db.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (melancholiai) [reply src/common/utils/db.ts:11]: removed.
using postgres for testing so there is no need for dates formation.

File: src/dumpMetadata/models/dumpMetadataManager.ts [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+  }
+
+  private getUrlHeader(): string {
+    const { protocol, host } = this.objectStorageConfig;
+    return `${protocol}://${host}`;
+  }
+
+  private parseDumpMetadataToDumpMetadataResponse(dumpMetadata: IDumpMetadata): DumpMetadataResponse {
+    const { bucket, ...restOfMetadata } = dumpMetadata;
+    const url = `${this.getUrlHeader()}/${bucket}/${restOfMetadata.name}`;
```
[the remark this answers is not in the document]
Comment (melancholiai) [reply src/dumpMetadata/models/dumpMetadataManager.ts:41]: done.

File: src/dumpMetadata/models/dumpMetadataManager.ts [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+    const dumpsMetadata = await this.repository.find(query);
+    return dumpsMetadata.map((dumpMetadata) => this.parseDumpMetadataToDumpMetadataResponse(dumpMetadata));
+  }
+
+  private getUrlHeader(): string {
+    const { protocol, host } = this.objectStorageConfig;
+    return `${protocol}://${host}`;
+  }
+
+  private parseDumpMetadataToDumpMetadataResponse(dumpMetadata: IDumpMetadata): DumpMetadataResponse {
```
[the remark this answers is not in the document]
Comment (melancholiai) [reply src/dumpMetadata/models/dumpMetadataManager.ts:39]: done.

File: tests/configurations/unit/jest.config.js [hunk 10]
```diff
@@ -12,4 +11,20 @@ module.exports = {
   setupFiles: ['<rootDir>/tests/configurations/jest.setup.js'],
   preset: 'ts-jest',
   testEnvironment: 'node',
+  collectCoverage: true,
+  collectCoverageFrom: [
+    '<rootDir>/src/dumpMetadata/models/*.ts',
+    '!<rootDir>/src/dumpMetadata/models/dumpMetadata.ts',
```
[the remark this answers is not in the document]
Comment (melancholiai) [reply tests/configurations/unit/jest.config.js:17]: it's needed in my env for some reason...

File: tests/helpers/index.ts [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+
+export const TOP_TO = faker.date.future(undefined, BOTTOM_FROM);
+
+export const createFakeUUID = (): string => {
+  return faker.random.uuid();
+};
+
+export const createFakeString = (): string => {
+  return faker.random.word();
+};
```
[the remark this answers is not in the document]
Comment (melancholiai) [reply tests/helpers/index.ts:30]: removed.

File: tests/helpers/index.ts (hunk 1 above)
Comment (melancholiai) [reply tests/helpers/index.ts:22]: nice.

File: tests/integration/dumpMetadata/helpers/db.ts (hunk 3 above)
Comment (melancholiai) [reply tests/integration/dumpMetadata/helpers/db.ts:10]: using this function two times :smile_cat:

File: tests/integration/dumpMetadata/helpers/requestSender.ts (hunk 4 above)
Comment (melancholiai) [reply tests/integration/dumpMetadata/helpers/requestSender.ts:21]: done.

File: tests/helpers/index.ts (hunk 2 above)
Comment (melancholiai) [reply tests/helpers/index.ts:64]: ResponeseResponeseResponeseResponeseResponeseResponese

**Comment by CptSchnitz** [comment]
@melancholiai you didnt push any new commits.

File: src/common/utils/db.ts [hunk 12]
```diff
@@ -1,22 +1,11 @@
...
-  DATE = 'yyyy-MM-dd',
-  DATETIME = 'yyyy-MM-dd HH:MM:ss',
-}
-
-export const moreThanOrEqualTimestamp = (date: Date, type: DateTypeFormat): FindOperator<string> => MoreThanOrEqual(format(date, type));
-
-export const lessThanOrEqualTimestamp = (date: Date, type: DateTypeFormat): FindOperator<string> => LessThanOrEqual(format(date, type));
-
-export const betweenTimestamps = (from: Date, to: Date, type: DateTypeFormat): FindOperator<string> => Between(format(from, type), format(to, type));
+export const CHARACTER_LENGTH_LIMIT = 100;
```
Comment (CptSchnitz) [inline src/common/utils/db.ts:5]: its name length limit, and shouldn't be generic.

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

File: src/common/utils/db.ts (hunk 12 above)
Comment (melancholiai) [reply src/common/utils/db.ts]: done.

**Review by CptSchnitz — APPROVED** [review]

**Review by CptSchnitz — APPROVED** [review]

**Pull request merged**